### ISSUE TITLE
`httpout` segfaults when processing requests

### ISSUE THREAD

**Issue by allanlei** [gpac/gpac#2695]
Thanks for reporting your issue. Please make sure these boxes are checked before submitting your issue - thank you!

- [x] I looked for a similar issue and couldn't find any.
- [x] I tried with the latest version of GPAC. Installers available at https://gpac.io/downloads/gpac-nightly-builds/
- [x] I give enough information for contributors to reproduce my issue (meaningful title, github labels, platform and compiler, command-line ...). I can share files anonymously with this dropbox: https://www.mediafire.com/filedrop/filedrop_hosted.php?drop=eec9e058a9486fe4e99c33021481d9e1826ca9dbc242a6cfaab0fe95da5e5d95

Detailed guidelines: https://gpac.io/bug-reporting/

---
Using `httpout` (via dash/hls output) results in Segmentation Fault on the first request.  [master c81887](https://github.com/gpac/gpac/commit/c81887c5d4e56b32efc3b6d81dff3c787dc41de4) seems to have this problem, but 2.2.1 does not

```
ffmpeg \
    -re -f lavfi -i testsrc=size=1280x720:rate=30:decimals=3 \
    -vf "drawtext=text='%{localtime\:%H.%M.%S}': x=0: y=0: fontsize=64 : fontcolor=white" \
    -force_key_frames 'expr:gte(t,n_forced*0.5)' -r:v 30 -keyint_min 30 -g 30 -c:v libx264 -b:v 1M -preset:v ultrafast -tune zerolatency \
    -movflags '+dash+delay_moov+skip_sidx+skip_trailer+frag_custom' \
    -f mpegts - | gpac \
        -i stdin \
        -logs=all@info \
        -log-utc \
        --reqlog=GET \
        --rdirs=. \
        -o 'http://127.0.0.1:8000/public-ll.m3u8:profile=live:dmode=dynamic'

curl 'http://127.0.0.1:8000/public-ll.m3u8'
```

* Version `2.2.1` will receive `200` response with proper content
* `master` will `Segmentation fault`

```
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [HTTPOut] Accepting new connection from 192.168.65.1
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Internal clock reference packet filtered - PID manifest_m3u8 clock ref 351000/90000 - type 1
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterdasherTasks] At 3414 Grabbed by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter dasher PID manifest_m3u8 unblocked (filter has 1 blocking pids)
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter dasher has only 1 / 2 blocked pids, requesting process task (0 queued)
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter dasher added to scheduler
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Thread 2695991328 Posted task 0xaaaae13b4270 Filter dasher::process (1 (1) pending, 1 process tasks) on secondary task list
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Thread 2695991328 posting filter task, scheduled_for_next_task 0
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterdasherTasks] At 3414 Released by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter httpout PID manifest_m3u8 (dasher) drop packet CTS 351000 SAP 0 Seek 0 - 0 packets remaining buffer 0 us
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter dasher PID manifest_m3u8 destroying packet CTS 351000
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter dasher PID manifest_m3u8 has 0 shared packets out
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterdasherTasks] At 3414 Grabbed by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterdasherTasks] At 3414 Released by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Internal clock reference packet filtered - PID P1V2 clock ref 3900/1000 - type 1
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex Filtermp4mxTasks] At 3414 Grabbed by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter mp4mx PID P1V2 unblocked (filter has 0 blocking pids)
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter mp4mx has only 0 / 1 blocked pids, requesting process task (0 queued)
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter mp4mx added to scheduler
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Thread 2695991328 Posted task 0xaaaae1335de0 Filter mp4mx::process (2 (1) pending, 1 process tasks) on secondary task list
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Thread 2695991328 posting filter task, scheduled_for_next_task 0
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex Filtermp4mxTasks] At 3414 Released by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter httpout PID P1V2 (mp4mx) drop packet CTS 3900 SAP 0 Seek 0 - 0 packets remaining buffer 0 us
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter mp4mx PID P1V2 destroying packet CTS 3900
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter mp4mx PID P1V2 has 0 shared packets out
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter httpout real-time reschedule in 2000 us (at 3415919 sys clock)
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter httpout process done
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterhttpoutTasks] At 3414 Grabbed by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterhttpoutTasks] At 3414 Released by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Thread 2695991328 re-posted task Filter httpout::process in secondary tasks (3 pending)
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Thread 2695991328 Waiting scheduler main semaphore
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Thread 2695991328 task#949 0xaaaae13b4270 executing Filter dasher::process (3 tasks pending, 1(1) process task queued)
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter dasher process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter dasher PID P1V2 has 4 shared packets out
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter dasher PID P1V2 sent packet DTS 405000 CTS 405000 SAP 0 seek 0 duration 3000 S/E 1/1 size 1366
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Dispatching packet from filter dasher to filter mp4mx - 1 packet in PID P1V2 buffer (33333 us buffer)
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterdasherTasks] At 3414 Grabbed by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterdasherTasks] At 3414 Released by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex Filtermp4mxTasks] At 3414 Grabbed by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter mp4mx skip post process task
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex Filtermp4mxTasks] At 3414 Released by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterdasherTasks] At 3414 Grabbed by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterdasherTasks] At 3414 Released by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterrfnaluTasks] At 3414 Grabbed by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter rfnalu PID P1V2 unblocked (filter has 0 blocking pids)
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter rfnalu has only 0 / 1 blocked pids, requesting process task (0 queued)
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter rfnalu added to scheduler
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Thread 2695991328 Posted task 0xaaaae133b4d0 Filter rfnalu::process (3 (1) pending, 1 process tasks) on secondary task list
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Thread 2695991328 posting filter task, scheduled_for_next_task 0
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterrfnaluTasks] At 3414 Released by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter dasher PID P1V2 (rfnalu) drop packet DTS 405000 CTS 405000 SAP 0 Seek 0 - 0 packets remaining buffer 0 us
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter dasher process done
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterdasherTasks] At 3414 Grabbed by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Filter] dasher removed from scheduler - blocking 1
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterdasherTasks] At 3414 Released by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterdasherTasks] At 3414 Grabbed by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterdasherTasks] At 3414 Released by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Thread 2695991328 task#949 0xaaaae13b4270 pushed to reservoir
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Thread 2695991328 Waiting scheduler main semaphore
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Thread 2695991328 task#950 0xaaaae1335de0 executing Filter mp4mx::process (3 tasks pending, 2(1) process task queued)
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter mp4mx process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterpinTasks] At 3414 Grabbed by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterpinTasks] At 3414 Released by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterhttpoutTasks] At 3414 Grabbed by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterhttpoutTasks] At 3414 Released by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex Filterm2tsdmxTasks] At 3414 Grabbed by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex Filterm2tsdmxTasks] At 3414 Released by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterrfnaluTasks] At 3414 Grabbed by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterrfnaluTasks] At 3414 Released by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterdasherTasks] At 3414 Grabbed by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterdasherTasks] At 3414 Released by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterhttpoutTasks] At 3414 Grabbed by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterhttpoutTasks] At 3414 Released by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex Filtermp4mxTasks] At 3414 Grabbed by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex Filtermp4mxTasks] At 3414 Released by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [MP4Mux] added sample DTS 276000 - prev DTS 273000 - prev size 993
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterdasherTasks] At 3414 Grabbed by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterdasherTasks] At 3414 Released by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter mp4mx PID P1V2 (dasher) drop packet DTS 402000 CTS 402000 SAP 0 Seek 0 - 1 packets remaining buffer 33333 us
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [MP4Mux] added sample DTS 279000 - prev DTS 276000 - prev size 1278
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterdasherTasks] At 3414 Grabbed by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter dasher PID P1V2 unblocked (filter has 0 blocking pids)
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter dasher has only 0 / 2 blocked pids, requesting process task (0 queued)
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter dasher added to scheduler
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Thread 2695991328 Posted task 0xaaaae13477c0 Filter dasher::process (3 (1) pending, 1 process tasks) on secondary task list
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Thread 2695991328 posting filter task, scheduled_for_next_task 0
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex FilterdasherTasks] At 3414 Released by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter mp4mx PID P1V2 (dasher) drop packet DTS 405000 CTS 405000 SAP 0 Seek 0 - 0 packets remaining buffer 0 us
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Filter mp4mx process done
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex Filtermp4mxTasks] At 3414 Grabbed by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Filter] mp4mx kept in scheduler blocking 0
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex Filtermp4mxTasks] At 3414 Released by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex Filtermp4mxTasks] At 3414 Grabbed by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - [Mutex Filtermp4mxTasks] At 3414 Released by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815161) - Thread 2695991328 re-posted task Filter mp4mx::process in secondary tasks (4 pending)
UTC 2023-11-27T05:43:35Z (TS 1701063815162) - Thread 2695991328 Waiting scheduler main semaphore
UTC 2023-11-27T05:43:35Z (TS 1701063815162) - [Mutex FilterhttpoutTasks] At 3415 Grabbed by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815162) - [Mutex FilterhttpoutTasks] At 3415 Released by thread Main Process
UTC 2023-11-27T05:43:35Z (TS 1701063815162) - Thread 2695991328: task httpout:process postponed for 5 ms (scheduled time 3415919 us, next task schedule 0 us)
UTC 2023-11-27T05:43:35Z (TS 1701063815167) - Thread 2695991328: task httpout:process schedule time 3415919 us reached (diff -3511 ms)
UTC 2023-11-27T05:43:35Z (TS 1701063815167) - Thread 2695991328 task#951 0xaaaae135f670 executing Filter httpout::process (4 tasks pending, 165(1) process task queued)
UTC 2023-11-27T05:43:35Z (TS 1701063815167) - Filter httpout process
av_interleaved_write_frame(): Broken pipe
Error writing trailer of pipe:: Broken pipe
frame=  103 fps= 30 q=15.0 Lsize=     344kB time=00:00:03.40 bitrate= 829.0kbits/s speed=0.995x    
video:313kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 9.779824%
[libx264 @ 0xaaaaf5f201c0] frame I:7     Avg QP:21.43  size: 25498
[libx264 @ 0xaaaaf5f201c0] frame P:96    Avg QP:20.46  size:  1484
[libx264 @ 0xaaaaf5f201c0] mb I  I16..4: 100.0%  0.0%  0.0%
[libx264 @ 0xaaaaf5f201c0] mb P  I16..4:  3.0%  0.0%  0.0%  P16..4:  6.9%  0.0%  0.0%  0.0%  0.0%    skip:90.0%
[libx264 @ 0xaaaaf5f201c0] final ratefactor: 21.34
[libx264 @ 0xaaaaf5f201c0] coded y,u,v intra: 2.8% 1.9% 1.7% inter: 0.5% 0.4% 0.3%
[libx264 @ 0xaaaaf5f201c0] i16 v,h,dc,p: 77%  9% 12%  2%
[libx264 @ 0xaaaaf5f201c0] kb/s:747.79
Conversion failed!
Segmentation fault
```

**Comment by jeanlf** [comment]
I cannot reproduce using latest master, could you double-check ?

**Comment by aureliendavid** [comment]
not reproducing either on latest master / ubuntu 22.04 / ffmpeg 4.4.2

**Comment by jeanlf** [comment]
closing, will reopen if/when we have enough info to reproduce
